applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp

```diff
@@ -613,9 +615,15 @@ double AdjointFiniteDifferencingBaseElement::GetPerturbationSizeModificationFact
 {
     KRATOS_TRY;
 
+    // For shape derivatives the size of the element (length, area, ...) is used as default perturbation size modification factor.
+    // Later on this value is multiplied with an user defined factor. This product is then used as final perturbation size for computing
+    // derivatives with finite differences.
     if(rDesignVariable == SHAPE)
     {
-        KRATOS_ERROR << "GetPerturbationSizeModificationFactor NOT_IMPLEMENTED" << std::endl;
+        const double domain_size = mpPrimalElement->GetGeometry().DomainSize();
+        KRATOS_DEBUG_ERROR_IF(domain_size <= std::numeric_limits<double>::epsilon())
```

Comment:
AFAIK the usage of `std::numeric_limits<double>::epsilon()` does not make sense in this context. Let's talk about this next week.

Reply:
I changed it like suggested.